daemon/cmd/policy.go

```diff
@@ -71,7 +71,7 @@ func (d *Daemon) initPolicy(
 		return fmt.Errorf("failed to create policy update trigger: %w", err)
 	}
 
-	d.monitorAgent.RegisterNewConsumer(authMonitor.AddAuthManager(auth.NewAuthManager(epMgr)))
+	d.monitorAgent.RegisterNewConsumer(authManager)
```

Comment:
Up for modularizing the monitor agent next so that auth.Cell could just register itself to the monitor? 
e.g. with an Invoke function. We don't need to use Provide() for things that are not providing an actual API but are just hooking into other places.

Reply:
yes, i'm happy to help and try to workout a first proposal. 

is there already an issue for this?

in the meantime i'm going to refactor some internals of the auth manager itself - e.g. by registering the different auth handlers via dedicated private cells which are using the hive/dig "value group" functionality. i tried to keep the scope of this PR as small as possible.